packages/openactive-integration-tests/test/features/cancellation/customer-requested-cancellation/implemented/atomic-cancel-test.js

```diff
@@ -0,0 +1,82 @@
+const { expect } = require('chai');
+const { FeatureHelper } = require('../../../../helpers/feature-helper');
+const { FlowStageRecipes, FlowStageUtils, OrderFeedUpdateFlowStageUtils, CancelOrderFlowStage } = require('../../../../helpers/flow-stages');
+const { itShouldReturnAnOpenBookingError } = require('../../../../shared-behaviours/errors');
+
+FeatureHelper.describeFeature(module, {
+  testCategory: 'cancellation',
+  testFeature: 'customer-requested-cancellation',
+  testFeatureImplemented: true,
+  testIdentifier: 'atomic-cancel',
+  testName: 'Successful booking and successful cancellation after atomic failed cancellation request',
+  testDescription: 'After a successful booking, and an unsuccessful but atomic cancellation request, successfully cancel, including checking the Orders feed.',
+  // Single Opportunity Criteria is overridden here as this test must have three Order Items
+  singleOpportunityCriteriaTemplate: opportunityType => [
+    {
+      opportunityType,
+      opportunityCriteria: 'TestOpportunityBookableCancellable',
+    },
+    {
+      opportunityType,
+      opportunityCriteria: 'TestOpportunityBookableCancellable',
+    },
+    {
+      opportunityType,
+      opportunityCriteria: 'TestOpportunityBookableCancellableOutsideWindow',
```

Comment:
So are you saying there should be an extra test criteria called `TestOpportunityBookableNotCancellableNotRefundable` and this test should use that instead of `TestOpportunityBookableCancellableOutsideWindow` ?

Separately is `TestOpportunityBookableNotCancellable` something that needs adding to the test-interface?

Reply:
Ah I just meant `TestOpportunityBookableNotCancellable` would be criteria where `allowCustomerCancellationFullRefund` is `false` OR the value of `latestCancellationBeforeStartDate` means that it currently can't be cancelled.

In both cases the cancellation should fail

Reply:
(Note `allowCustomerCancellationFullRefund` is now in the Orders feed 🙌  so can be used in the criteria)

Reply:
`TestOpportunityBookableNotCancellable` is currently not a criteria, so that will have to be added into the `test-interface`.

Sorry, I should have been clearer, I understand that `TestOpportunityBookableNotCancellable` is a superset of those two conditions, I'm just not very familiar with how to define that superset in Test Suite, or `test-interface`. Does it require an additional criteria `TestOpportunityBookableNotCancellableNotRefundable`?

Reply:
Oh I see sorry, the criteria is just an emum in the test interface so there's no magic there, just a new `TestOpportunityBookableNotCancellable`. No need to defined `TestOpportunityBookableNotCancellableNotRefundable` separately. It's really just an enum to represent the constraints.

In the criteria definitions in the test suite itself it might be worth subclassing TestOpportunityBookableCancellableOutsideWindow as this is adding an additional constraint on top of that

Reply:
1) Criteria are based on the `offer` or on the `opportunity`. How do I specify a criteria based on a field that is in the `acceptedOffer` of the `Order` as this isn't available at the start of the booking flow?

2) If `TestOpportunityBookableNotCancellable` were to subclass `TestOpportunityBookableCancellableOutsideWindow`, wouldn't the criteria be allowCustomerCancellationFullRefund is false AND the value of latestCancellationBeforeStartDate , not OR ?

Reply:
1. What's in the acceptedOffer of the Order must also be in the open data feeds. If `allowCustomerCancellationFullRefund` is not yet in the feeds in ref impl it should be added 

2. You're totally right! Sorry! Yes not a subclass.

Reply:
1. And for full avoidance of doubt `allowCustomerCancellationFullRefund` should be on `Offer` right? The change was that it used to be on the `OrderItem` and therefore wasn't available before booking? Is that right?

Reply:
(I think the reason I'm confused is that there are examples in the BookingSpec as this is a recent change.)

Reply:
yes that's exactly right